lib/puppet/provider/service/smf.rb

```diff
@@ -135,8 +143,8 @@ def status
       states = service_states
       state = states[1] == "-" ? states[0] : states[1]
     rescue Puppet::ExecutionFailure
-      info "Could not get status on service #{self.name}"
-      return :stopped
+      debug "Could not get status on service #{self.name} #{$!}"
+      return :absent
```

Comment:
Strictly speaking I'm not sure that we should ever reach this code anymore but I'm not sure. Is there a way to get here without discovering the service doesn't exist via exists? 

If it is reached it also adds the complexity that :absent isn't a valid service state. 
Such that if status is :absent but the requested state is :stopped the end result is a change on every execution that fails to take an absent service to a stopped state.

However, I think that is all addressed via the addition of exists? which then seems to "do the right thing(TM)"